components/constellation/constellation.rs

```diff
@@ -2092,6 +2102,17 @@ where
             }
         }
 
+        if let Some(webgpu) = self.webgpu.as_ref() {
+            debug!("Exiting WebGPU thread.");
+            let (sender, receiver) = ipc::channel().expect("Failed to create IPC channel!");
+            if let Err(e) = webgpu.exit(sender) {
```

Comment:
I think it would be better to move the channel creation and `recv` to inside `exit`.